src/node.cc

```diff
@@ -93,10 +94,26 @@ std::ostream &operator<<(std::ostream &stream, const AbstractNode &node)
 // Do we have an explicit root node (! modifier)?
 AbstractNode *find_root_tag(AbstractNode *n)
 {
-  for(auto v : n->children) {
-    if (v->modinst->tag_root) return v;
-    if (auto vroot = find_root_tag(v)) return vroot;
-  }
-  return NULL;
-}
+	std::vector<AbstractNode*> rootTags;
+
+	std::function <void (AbstractNode *n)> find_root_tags = [&] (AbstractNode *n)
```

Comment:
Okay.

By the way, I was thinking of restricting the capture list in which case auto will not work.

```
 std::function <void (AbstractNode *n)> find_root_tags =  find_root_tags = [&rootTags, &find_root_tags] (AbstractNode *n)

```